chain/pra/receiver_relay.go

```diff
@@ -0,0 +1,256 @@
+package pra
+
+import (
+	"github.com/icon-project/btp/chain/icon"
+	"github.com/icon-project/btp/chain/pra/substrate"
+	"github.com/icon-project/btp/common/log"
+	"github.com/icon-project/btp/common/mta"
+)
+
+type relayReceiver struct {
+	c               substrate.SubstrateClient
+	bmvC            icon.PraBmvClient
+	bmvStatus       icon.PraBmvStatus
+	expectMtaHeight uint64 // keep track of MtaHeight per paraFinalityProof
+	expectSetId     uint64
+	store           *mta.ExtAccumulator
+	log             log.Logger
+}
+
+func NewRelayReceiver(opt receiverOptions, log log.Logger) relayReceiver {
+	rC := relayReceiver{log: log}
+	rC.c, _ = substrate.NewSubstrateClient(opt.RelayEndpoint)
+	rC.bmvC = icon.NewPraBmvClient(opt.DstEndpoint, opt.PraBmvAddress.ContractAddress(), log)
+	rC.bmvStatus = rC.bmvC.GetPraBmvStatus()
+	rC.prepareDatabase(int64(opt.RelayOffSet), opt.AbsBaseDir(), opt.RelayBtpAddress.NetworkAddress())
+	return rC
+}
+
+func (r *relayReceiver) findParasInclusionCandidateIncludedHead(from uint64, paraHead substrate.SubstrateHash, paraChainId substrate.SubstrateParachainId) (*substrate.SubstrateHeader, *substrate.SubstrateHash) {
+	r.log.Debugf("findParasInclusionCandidateIncludedHead: from %d paraHead: %s", from, paraHead.Hex())
+	foundEvent := false
+	blockNumber := from
+	for !foundEvent {
```

Comment:
the foundEvent never be changed?

Reply:
yes, the foundEvent never be changed if and only if parachain doesn't produce block anymore.

Reply:
I mean , we do we need this varaiable ?

Reply:
No, we don't need, I remved this variable